socketshark/subscription.py

```diff
@@ -173,6 +173,35 @@ async def periodic_authorizer(self):
                 )
                 await self.self_unsubscribe(e.error)
 
+    async def send_heartbeat(self):
+        await self.perform_service_request('on_heartbeat')
+
+    async def periodic_heartbeat(self):
+        period = self.service_config['heartbeat_period']
+        self.session.trace_log.debug(
+            'initializing periodic heartbeat task',
+            subscription=self.name,
+            period=period,
+        )
+        while True:
+            await asyncio.sleep(period)
+            try:
+                self.session.log.debug(
```

Comment:
I wonder if it should use `trace_log`, but `periodic_authorizer` does not. I guess it's not super clear which logger to use.

Reply:
yes - left it the same as periodic_authorizer. LMK if I should change both to `trace_log`.